source/uwp/Renderer/lib/RenderedAdaptiveCard.cpp

```diff
@@ -124,6 +214,34 @@ namespace AdaptiveNamespace
         m_originatingCard = value;
     }
 
+    void RenderedAdaptiveCard::SetOriginatingHostConfig(_In_ ABI::AdaptiveNamespace::IAdaptiveHostConfig* value)
+    {
+        m_originatingHostConfig = value;
+    }
+
+    HRESULT RenderedAdaptiveCard::AddInlineShowCard(_In_opt_ IAdaptiveActionSet* actionSet,
+                                                    _In_ IAdaptiveShowCardAction* showCardAction,
+                                                    _In_ ABI::Windows::UI::Xaml::IUIElement* showCardFrameworkElement) try
+    {
+        InternalId actionSetId;
+        if (actionSet != nullptr)
+        {
+            // If this action in in an actionset, store the action set id
+            ComPtr<AdaptiveNamespace::AdaptiveActionSet> actionSetImpl =
+                PeekInnards<AdaptiveNamespace::AdaptiveActionSet>(actionSet);
+            actionSetId = actionSetImpl->GetInternalId();
+        }
+
+        ComPtr<AdaptiveNamespace::AdaptiveShowCardAction> showCardImpl =
+            PeekInnards<AdaptiveNamespace::AdaptiveShowCardAction>(showCardAction);
+        InternalId showCardActionId = showCardImpl->GetInternalId();
+
+        m_showCards.emplace(std::make_pair(showCardActionId, std::make_pair(actionSetId, showCardFrameworkElement)));
```

Comment:
>std::make_pair(showCardActionId, std::make_pair(actionSetId, showCardFrameworkElement)) [](start = 28, length = 87)

I think it's fine if you prefer this style (it's more explicit), but you should also be able to pull it off with an initializer list:

```
m_showCards.emplace({showCardActionId, {actionSetId, showCardFrameworkElement}});
```

Reply:
That approach doesn't seem to immediately compile.  Possibly i could get it to work if I futzed with it enough, but I'm going to leave it as is for now.

---
In reply to: [261749093](https://github.com/Microsoft/AdaptiveCards/pull/2494#discussion_r261749093) [](ancestors = 261749093)

Reply:
fine by me :)